src/scripts/main.js

```diff
@@ -1,3 +1,44 @@
 'use strict';
 
-// write code here
+const ul = document.querySelector('.tree');
+
+// We create a span in li that contains a list
+function addSpan(elem) {
+  const elementChildren = [...elem.children];
+
+  if (elementChildren.length === 0) {
+    return;
+  }
+
+  if (elementChildren.length === 1) {
+    const span = document.createElement('span');
+    const textLi = elem.childNodes[0].textContent.trim();
+
+    span.textContent = textLi;
+    elem.prepend(span);
+    elem.childNodes[1].remove();
+  }
+
+  for (const el of elementChildren) {
+    addSpan(el);
+  }
+}
+
+addSpan(ul);
+
+// We hide ul when clicking on span
+document.addEventListener('click', (e) => {
+  const target = e.target.closest('span');
+
+  if (!target) {
+    return;
+  }
+
+  const list = target.nextSibling;
```

Comment:
The variable 'list' is assigned 'target.nextSibling', which may not always be the correct element. Ensure that 'target.nextSibling' is the expected 'ul' element. Consider checking the node type or using a more reliable method to select the sibling 'ul' element.